File: sorts/basic_count_sort.py
class Solution:
    def countSort(self, nums, max_value) -> None:
      range_value = max_value - min_value + 1
      # build empty count array
      count = [0] * range_value
  
      # Count occurences
      for num in nums:
          count[num - min_value] += 1
      

      # Reconstruct the sorted_arr
      index = 0
      for i in range(range_value):
          while count[i] > 0:
            nums[index] = i + min_value
            index += i
            count[i] -= 1
      return nums
      

    def sortColors(self, nums: List[int]) -> None:
        
      return self.countSort(nums, min(nums), max(nums))

File: sorts/basic_count_sort.rb

def counting_sort(arr, min_value, max_value)
    range_value = max_value - min_value + 1
    # build empty count array
    count = Array.new(range_value, 0)

    # Count occurences
    arr.each do |num|
        count[num - min_value] += 1
    end

    # Reconstruct the sorted_arr
    index = 0
    (0...range_value).each do |i|
        while count[i] > 0
            arr[index] = i + min_value
            index += 1
            count[i] -= 1
        end
    end

    arr
end

def sort_colors(nums)
    return nums if nums.empty?
    counting_sort(nums, nums.min, nums.max)
end